### PR TITLE
ROS Melodic compatibility for aruco_pose package

### PR DESCRIPTION
**Pull request by sfalexrog**
ROS Melodic dropped the OpenCV package and relies on the system package version. For supported systems this translates to OpenCV 3.2.0, which has API incompatibilities in aruco package.

This P/R introduces checks for OpenCV version and reroutes API access to match currently used OpenCV.

### CODE REVIEW

File: aruco_pose/src/compat.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		}
+	}
+
+	// create output
+	Mat(objPnts).copyTo(objPoints);
+	Mat(imgPnts).copyTo(imgPoints);
+#endif
+}
+
+}
```
Comment (okalachev) [inline aruco_pose/src/compat.cpp]: Trailing newline needed

File: aruco_pose/src/compat.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ *
+ * Authors: Oleg Kalachev <okalachev@gmail.com>, Alexey Rogachevskiy <sfalexrog@gmail.com>
+ *
+ * Distributed under MIT License (available at https://opensource.org/licenses/MIT).
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ */
+
+/*
+ * Code is based on https://github.com/opencv/opencv_contrib, which is distributed
```
Comment (okalachev) [inline aruco_pose/src/compat.cpp]: I'm not sure something in this files is based on that repo.

Comment (sfalexrog) [reply aruco_pose/src/compat.cpp]: The code in this file is directly lifted from opencv_contrib, see: https://github.com/opencv/opencv_contrib/blob/master/modules/aruco/src/aruco.cpp#L1079-L1109

File: aruco_pose/src/compat.cpp (hunk 1 above)
Comment (sfalexrog) [reply aruco_pose/src/compat.cpp]: Done

**Comment by sfalexrog** [comment]
Partially superseded by #156 (though it lacks tests for `aruco_pose` package, so not closing this yet)

**Comment by sfalexrog** [comment]
Superseded completely by #156.

**Pull request closed without merging**